setup/bindir/cloud-set-guest-sshkey-password-userdata-configdrive.in

```diff
@@ -0,0 +1,507 @@
+            return 0
+        fi
+    else
+        outputLog "Cannot determine change of password"
+    fi
+    outputLog "Changing password for user $username"
+    if command -v chpasswd &> /dev/null
+    then
+        echo "$username":"$password" | chpasswd
+    elif command -v usermod &> /dev/null && command -v mkpasswd &> /dev/null
+    then
+        usermod -p "$(mkpasswd -m SHA-512 "$password")" "$username"
+    elif command -v pw &> /dev/null
+    then
+        echo "$password" | pw mod user "$username" -h 0
+    else
+        outputLog "Failed to change password for user $username"
+        return 1
+    fi
+    outputLog "Successfully changed password for user $username"
+    if [ $newMd5 != 'N/A' ]
+    then
+        echo "$newMd5" > "$homeDir"/.password.md5
+        chmod 600 "$homeDir"/.password.md5
+        chown "$username": "$homeDir"/.password.md5
+    fi
+    return 0
+}
+
+function sendAckToUserDataServer(){
+    userDataServer=$1
+    userDataServerPort=$2
+    outputLog "Sending acknowledgment to userdata server at $userDataServer"
+    if ! curl --silent --connect-timeout 20 --retry 3 --header "DomU_Request: saved_password" "$userDataServer":"$userDataServerPort" &> /dev/null
+    then
+        outputLog "Failed to sent acknowledgment to userdata server at $userDataServer"
+        return 1
+    fi
+    outputLog "Successfully sent acknowledgment to userdata server at $userDataServer"
+    return 0
+}
+
+function getPublicKeyFromUserDataServer(){
+    userDataServer=$1
+    outputLog "Sending request to userdata server at $userDataServer to get public key"
+    if ! reponse=$(curl --silent --connect-timeout 20 --retry 3 http://"$userDataServer"/latest/public-keys)
+    then
+        outputLog "Failed to get public key from userdata server"
+        return 2
+    fi
+    outputLog "Got response from userdata server at $userDataServer"
+    if [ -z "$reponse" ]
+    then
+        outputLog "Did not receive any public keys from userdata server"
+        return 1
+    fi
+    outputLog "Successfully get public key from userdata server"
+    echo "$reponse"
+    return 0
+}
+
+function setPublicKey(){
+    username=$1
+    homeDir=$2
+    publicKey=$3
+    outputLog "Applying public key for $username"
+    sshDir=$homeDir/.ssh
+    authorizedKeysFile=$sshDir/authorized_keys
+
+    if [ ! -d "$sshDir" ]
+    then
+        outputLog ".ssh directory for $username not found, creating .ssh directory"
+        mkdir "$sshDir"
+    fi
+
+    if [ ! -f "$authorizedKeysFile" ]
+    then
+        outputLog "authorized_keys file for $username not found, creating authorized_keys file"
+        touch "$authorizedKeysFile"
+    fi
+    if grep "$(echo "$publicKey" | awk '{print $2}')" "$authorizedKeysFile" > /dev/null
+    then
+        outputLog "No need to update authorized_keys file"
+        return 0
+    fi
+    outputLog "Writing public key in authorized_keys file"
+    echo "$publicKey" >> "$authorizedKeysFile"
```

Comment:
if use ">>", old ssh key will not be removed

Reply:
I think it is expected behavior as I get the logic from old script:
https://github.com/apache/cloudstack/blob/07722ce016c275ad864254db4008e1a70610596b/setup/bindir/cloud-set-guest-sshkey.in#L85

Reply:
@aleskxyz 
can you use the same code ?
```
sed -i "/ cloudstack@apache.org$/d" $authorized
echo "$publickey cloudstack@apache.org" >> $authorized
```

Reply:
Yea! Now the script overwrites ssh keys injected by CloudStack only.